Foundation/FileManager.swift

```diff
@@ -1441,73 +1445,53 @@ open class FileManager : NSObject {
-            }
-            
-            if stream != nil {
-                defer {
-                    fts_close(stream)
+                if (URL(fileURLWithPath: itemPath).hasDirectoryPath) {
+                  if rmdir(itemPath) == -1 {
+                    throw _NSErrorWithErrno(errno, reading: false, path: itemPath)
+                  }
+                } else {
+                  if unlink(itemPath) == -1 {
+                    throw _NSErrorWithErrno(errno, reading: false, path: itemPath)
+                  }
                 }
-
-                while let current = fts_read(stream)?.pointee {
-                    let itemPath = string(withFileSystemRepresentation: current.fts_path, length: Int(current.fts_pathlen))
-                    guard alreadyConfirmed || shouldRemoveItemAtPath(itemPath, isURL: isURL) else {
-                        continue
-                    }
-                    
-                    do {
-                        switch Int32(current.fts_info) {
-                        case FTS_DEFAULT, FTS_F, FTS_NSOK, FTS_SL, FTS_SLNONE:
-                            if unlink(current.fts_path) == -1 {
-                                throw _NSErrorWithErrno(errno, reading: false, path: itemPath)
-                            }
-                        case FTS_DP:
-                            if rmdir(current.fts_path) == -1 {
-                                throw _NSErrorWithErrno(errno, reading: false, path: itemPath)
-                            }
-                        case FTS_DNR, FTS_ERR, FTS_NS:
-                            throw _NSErrorWithErrno(current.fts_errno, reading: false, path: itemPath)
-                        default:
-                            break
-                        }
-                    } catch {
-                        if !shouldProceedAfterError(error, removingItemAtPath: itemPath, isURL: isURL) {
-                            throw error
-                        }
-                    }
+              } catch {
+                if !shouldProceedAfterError(error, removingItemAtPath: itemPath, isURL: true) {
+                  throw error
                 }
-            } else {
-                let _ = _NSErrorWithErrno(ENOTEMPTY, reading: false, path: path)
+              }
             }
-        } else if errno != ENOTDIR {
-            throw _NSErrorWithErrno(errno, reading: false, path: path)
-        } else if _fileSystemRepresentation(withPath: path, { unlink($0) != 0 }) {
-            throw _NSErrorWithErrno(errno, reading: false, path: path)
+        } else if _fileSystemRepresentation(withPath: path, {path in
+                                            #if os(Windows)
+                                            return _NS_unlink(path) != 0
+                                            #else
+                                            return unlink(path) != 0
+                                            #endif
```

Comment:
This can also be `unlink(path)` now right?

Reply:
eeeyup